packages/components/src/components/hds/tabs/index.ts

```diff
@@ -168,52 +190,52 @@ export default class HdsTabsIndexComponent extends Component {
   }
 
   @action
-  onKeyUp(tabIndex, event) {
-    const leftArrow = 37;
-    const rightArrow = 39;
-    const enterKey = 13;
-    const spaceKey = 32;
+  onKeyUp(tabIndex: number, event: KeyboardEvent) {
+    const leftArrow = 'ArrowLeft';
+    const rightArrow = 'ArrowRight';
+    const enterKey = 'Enter';
+    const spaceKey = ' ';
 
-    if (event.keyCode === rightArrow) {
+    if (event.key === rightArrow) {
       const nextTabIndex = (tabIndex + 1) % this.tabIds.length;
       this.focusTab(nextTabIndex, event);
-    } else if (event.keyCode === leftArrow) {
+    } else if (event.key === leftArrow) {
       const prevTabIndex =
         (tabIndex + this.tabIds.length - 1) % this.tabIds.length;
       this.focusTab(prevTabIndex, event);
-    } else if (event.keyCode === enterKey || event.keyCode === spaceKey) {
+    } else if (event.key === enterKey || event.key === spaceKey) {
       this.selectedTabIndex = tabIndex;
     }
     // scroll selected tab into view (it may be out of view when activated using a keyboard with `prev/next`)
-    this.tabNodes[this.selectedTabIndex].parentNode.scrollIntoView({
+    (
+      this.tabNodes[this.selectedTabIndex]?.parentNode as HTMLElement
+    ).scrollIntoView({
       behavior: 'smooth',
       block: 'nearest',
       inline: 'nearest',
     });
```

Comment:
I would generally prefer type checking over type casting
```suggestion
    const parentNode = this.tabNodes[this.selectedTabIndex]?.parentNode;
    if (parentNode instanceof HTMLElement) {
      parentNode.scrollIntoView({
        behavior: 'smooth',
        block: 'nearest',
        inline: 'nearest',
      });
    }
}```

Reply:
+1